### PR TITLE
Release 5 - Caution form notification

### PR DESCRIPTION
**Pull request by SaroshHashmi**

### CODE REVIEW

File: AirNZ.CS.CRM.WebResources/Jscript/Case.js [hunk 1]
```diff
@@ -156,5 +168,63 @@ AirNZ.Case.FormEvents = {
...
+                function success(result)
+                {
+                    let Warning = false;
+                    let Ban = false;
+
+                    if (result.entities.length > 0)
+                    {
+                        for (let i = 0; i < result.entities.length; i++)
+                        {
+                            if (result.entities[i].airnz_type == CautionType.Warning) {
```
Comment (brettrattey) [inline AirNZ.CS.CRM.WebResources/Jscript/Case.js:198]: Pls use === over == where loose comparison is not required. 

_The == operator performs a loose equality comparison that performs type coercion if necessary to make the comparison possible. The === operator, on the other hand, performs a strict equality comparison that does not perform type coercion and requires the operands to have the same type (as well as the same value).__

File: AirNZ.CS.CRM.WebResources/Jscript/Case.js [hunk 2]
```diff
@@ -156,5 +168,63 @@ AirNZ.Case.FormEvents = {
...
+                                Warning = true;
+                            }
+                            else if(result.entities[i].airnz_type == CautionType.Ban)
```
Comment (brettrattey) [inline AirNZ.CS.CRM.WebResources/Jscript/Case.js:201]: Pls use === over ==

File: AirNZ.CS.CRM.WebResources/Jscript/Caution.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+    //Set Notification if contact has associated Caution
+    SetEndDate: function (executionContext) {
+        'use strict';
+
+        const formContext = executionContext.getFormContext();
+        let endDate = formContext.getAttribute(EndDate).getValue();
+        const startDate = formContext.getAttribute(StartDate).getValue();
+
+        if (endDate == null && startDate != null)
```
Comment (brettrattey) [inline AirNZ.CS.CRM.WebResources/Jscript/Caution.js:27]: Pls use ===  or !== over == and !=

File: AirNZ.CS.CRM.WebResources/Jscript/Contact.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+                    let Ban = false;
+
+                    if (result.entities.length > 0)
+                    {
+                        for (let i = 0; i < result.entities.length; i++)
+                        {
+                            if (result.entities[i].airnz_type == CautionType.Warning) {
+                                Warning = true;
+                            }
+                            else if(result.entities[i].airnz_type == CautionType.Ban)
```
Comment (brettrattey) [inline AirNZ.CS.CRM.WebResources/Jscript/Contact.js:56]: Pls use === over ==

File: AirNZ.CS.CRM.WebResources/Jscript/Email.js [hunk 5]
```diff
@@ -111,6 +111,33 @@ AirNZ.Email.FormEvents = {
...
+        let req = new XMLHttpRequest();
+        req.open("POST", requestURL, true);
+        req.setRequestHeader("OData-MaxVersion", "4.0");
+        req.setRequestHeader("OData-Version", "4.0");
+        req.setRequestHeader("Accept", "application/json");
+        req.setRequestHeader("Content-Type", "application/json; charset=utf-8");
+        req.onreadystatechange = function () {
+
+            if (this.readyState === 4) {
+                if (this.status == 200) {
```
Comment (brettrattey) [inline AirNZ.CS.CRM.WebResources/Jscript/Email.js:130]: Pls use === over ==

File: AirNZ.CS.CRM.WebResources/Jscript/Case.js (hunk 1 above)
Comment (SaroshHashmi) [reply AirNZ.CS.CRM.WebResources/Jscript/Case.js]: done

File: AirNZ.CS.CRM.WebResources/Jscript/Case.js (hunk 2 above)
Comment (SaroshHashmi) [reply AirNZ.CS.CRM.WebResources/Jscript/Case.js]: done

File: AirNZ.CS.CRM.WebResources/Jscript/Caution.js (hunk 3 above)
Comment (SaroshHashmi) [reply AirNZ.CS.CRM.WebResources/Jscript/Caution.js]: done

File: AirNZ.CS.CRM.WebResources/Jscript/Contact.js (hunk 4 above)
Comment (SaroshHashmi) [reply AirNZ.CS.CRM.WebResources/Jscript/Contact.js]: done

File: AirNZ.CS.CRM.WebResources/Jscript/Email.js (hunk 5 above)
Comment (SaroshHashmi) [reply AirNZ.CS.CRM.WebResources/Jscript/Email.js]: done

**Pull request merged**